src/main/java/org/testng/internal/PackageUtils.java

```diff
@@ -83,68 +84,92 @@ public static void addClassLoader(final ClassLoader loader) {
       }
     }
 
+    ParamContainer container = new ParamContainer(included,excluded,vResult);
```

Comment:
The only reason why i had to resort to doing this was because sonar was complaining when it saw a method which had more than 7 params. So i just wrapped these three lists into the container.